lightly_studio/src/lightly_studio/api/routes/api/text_embedding.py

```diff
@@ -35,9 +38,15 @@ def embed_text(
     ] = None,
 ) -> list[float]:
     """Retrieve embeddings for the input text."""
+    # TODO(Jonas, 12/2025): Remove this hack after model_id is provided from frontend
+    # This is a hack, since at the moment, no valid embedding_model_id is passed from the frontend.
+    # so we fetch the root_dataset_id, which will be used inside embed_text to get the default model
+    # for this dataset.
+    root_dataset = dataset_resolver.get_root_dataset(session=session)
+    dataset_id = root_dataset.dataset_id
     try:
         text_embeddings = embedding_manager.embed_text(
-            TextEmbedQuery(query_text, embedding_model_id)
+            dataset_id=dataset_id, text_query=TextEmbedQuery(query_text, embedding_model_id)
```

Comment:
Minor: It would be nice to add the kwargs if you are touching the line:
```suggestion
            dataset_id=dataset_id, text_query=TextEmbedQuery(text=query_text, embedding_model_id=embedding_model_id)
```